source/initial_temperature/patch_on_S40RTS.cc

```diff
@@ -0,0 +1,206 @@
+                             "The maximum depth of the Vs ascii grid. The model will read in  "
+                             "Vs from S40RTS below this depth.");
+          prm.declare_entry ("Smoothing length scale", "200000.0",
+                             Patterns::Double (0),
+                             "The depth range (above maximum grid depth) over which to smooth. "
+                             "The boundary is smoothed using a depth weighted combination of Vs "
+                             "values from the ascii grid and S40RTS at each point in the region of smoothing.");
+
+          Utilities::AsciiDataBase<dim>::declare_parameters(prm,
+                                                            "$ASPECT_SOURCE_DIR/data/initial-temperature/patch-on-S40RTS/test/",
+                                                            "upper_shell_3d.txt");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+    template <int dim>
+    void
+    PatchOnS40RTS<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial temperature model");
+      {
+        prm.enter_subsection ("Patch on S40RTS");
+        {
+          max_grid_depth           = prm.get_double ("Maximum grid depth");
+          smoothing_length_scale   = prm.get_double ("Smoothing length scale");
+
+          Utilities::AsciiDataBase<dim>::parse_parameters(prm);
+
+        }
+        prm.leave_subsection ();
+      }
+      prm.leave_subsection ();
+      s40rts.initialize_simulator (this->get_simulator());
+
+      // Note: parse_parameters will call initialize for us
+      s40rts.parse_parameters(prm);
+
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace InitialTemperature
+  {
+    ASPECT_REGISTER_INITIAL_TEMPERATURE_MODEL(PatchOnS40RTS,
+                                              "patch on S40RTS",
+                                              "Implementation of a model in which the initial "
+                                              "temperature is derived from a file containing shear wave velocity perturbations "
+                                              "in ascii format (e.g. a high resolution upper mantle tomography) "
+                                              "combined with S40RTS. Note the required format of the "
+                                              "input ascii input data: The first lines may contain any number of comments "
+                                              "if they begin with '#', but one of these lines needs to "
+                                              "contain the number of grid points in each dimension as "
+                                              "for example '# POINTS: 3 3 3'. "
+                                              "The order of the data columns has to be "
+                                              " `x', `y', `z', 'Temperature [K]' in a 3d model, which means that "
```

Comment:
The 'Temperature' has to be replaced by 'Vs anomaly' (or another suitable name) and the corresponding unit.